gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt

```diff
@@ -603,13 +594,7 @@ class KspGradleSubplugin @Inject internal constructor(private val registry: Tool
             project.files(kotlinOutputDir).builtBy(kspTaskProvider),
             project.files(javaOutputDir).builtBy(kspTaskProvider),
         )
-        if (kotlinCompilation is KotlinCommonCompilation) {
-            // Do not add generated sources to common source sets.
-            // They will be observed by downstreams and violate current build scheme.
-            kotlinCompileProvider.configure { it.source(*generatedSources) }
-        } else {
-            kotlinCompilation.defaultSourceSet.kotlin.srcDirs(*generatedSources)
```

Comment:
The reason for adding generated sources to `kotlinCompilation.defaultSourceSet` and all corresponding workarounds is IDE support, which is not covered in KSP's tests unfortunately. Without this, users would add the output folder of KSP to the source set and suffer from the same (if not more) issue.

I got a couple of alternatives but they are apparently beyond the scope of this PR.
1. Work with KGP / JetBrains to get an API for registering generated sources, like those provided by AGP.
2. Implement IDE plugins for KSP.

Are there easier solutions?

Reply:
Related: https://github.com/gradle/gradle/issues/32053